lib/active_merchant/billing/gateways/card_connect.rb

```diff
@@ -175,13 +190,19 @@ def add_invoice(post, options)
       end
 
       def add_payment(post, payment)
-        post[:name] = payment.name
-        if card_brand(payment) == 'check'
-          add_echeck(post, payment)
+        if payment.is_a?(String)
+          account_id, profile_id = payment.split('||')
```

Comment:
This is still splitting on `||`

Reply:
Well that'll teach me not to run remote tests. Fixed (along with the corresponding test).